tartufo/scanner.py

```diff
@@ -683,3 +683,49 @@ def chunks(self):
         )
         for blob, file_path in self._iter_diff_index(diff_index):
             yield types.Chunk(blob, file_path, {})
+
+
+class FolderScanner(ScannerBase):
+    """Used to scan a folder."""
+
+    folder_path: str
+
+    def __init__(
+        self,
+        global_options: types.GlobalOptions,
+        target: str,
+    ) -> None:
+        """Used for scanning a folder.
+
+        :param global_options: The options provided to the top-level tartufo command
+        :param target: The local filesystem path to scan
+        """
+        self.target = target
+        super().__init__(global_options)
+
+    @property
+    def chunks(self) -> Generator[types.Chunk, None, None]:
+        """Yield the individual file changes currently staged for commit.
+
+        :rtype: Generator[Chunk, None, None]
+        """
+
+        for blob, file_path in self._iter_folder():
+            if self.should_scan(file_path):
+                yield types.Chunk(blob, file_path, {})
+
+    def _iter_folder(self) -> Generator[types.Tuple, None, None]:
+        folder_path = pathlib.Path(self.target)
+        for file_path in folder_path.rglob("**/*"):
+            relative_path = file_path.relative_to(folder_path)
+            if file_path.is_file() and self.should_scan(str(relative_path)):
+                with file_path.open("rb") as fhd:
+                    data = fhd.read()
```

Comment:
This should probably be in a try/except block to fail gracefully (display an error message?) when we can't open a file.

Reply:
Sure. I know tartufo reads many files. Can you point me to an example of one that wraps with a try/except?

Reply:
nvm, I found one. Using

```
click.FileError(filename=str(config_path / value), hint=str(exc))
```

Reply:
This example unfortunately only handles file not found and not permissions errors. Ok to capture `OSError`?

Reply:
Yeah I think you would basically want something like
```python
try:
    with file_path.open("rb") as fhd:
        data = fhd.read()
except OSError as exc:
    log.warning("Could not open file %s. (%s)", file_path, exc)
```

Reply:
The `OSError` class serves as the base class for all the exception types you would need to capture here, including permissions, file not found, etc. https://docs.python.org/3/library/exceptions.html#os-exceptions